### PR TITLE
Recipe edit bug hunt

### PR DESCRIPTION
**Pull request by switzerb**
This PR addresses:
- https://github.com/folded-ear/gobrennas-client/issues/98
- https://github.com/folded-ear/gobrennas-client/issues/100
- https://github.com/folded-ear/gobrennas-client/issues/101
- https://github.com/folded-ear/gobrennas-client/issues/102

### CODE REVIEW

File: src/features/RecipeEdit/RecipeAddController.tsx [hunk 1]
```diff
@@ -6,10 +6,13 @@ import { useCreateRecipe } from "data/hooks/useCreateRecipe";
 import { useGetAllLabels } from "data/hooks/useGetAllLabels";
 import ClientId from "util/ClientId";
 import { useHistory } from "react-router-dom";
+import { ApolloError } from "@apollo/client";
+import { Alert } from "@mui/material";
 
 export const RecipeAddController = () => {
     const { data: labelList } = useGetAllLabels();
     const { createRecipe } = useCreateRecipe();
+    const [error, setError] = React.useState<ApolloError | null>(null);
```
Comment (barneyb) [inline src/features/RecipeEdit/RecipeAddController.tsx:15]: Wouldn't getting `error` from `useCreateRecipe` do basically the same thing, without needing to manually manage state and promises, and support "you tried again and it worked" for free?

File: src/features/RecipeEdit/RecipeAddController.tsx [hunk 2]
```diff
@@ -36,12 +39,16 @@ export const RecipeAddController = () => {
...
-        createRecipe(recipe).then((result) => {
-            const id = result.data?.library?.createRecipe.id;
-            id
-                ? history.push(`/library/recipe/${id}`)
-                : history.push(`/library`);
-        });
+        createRecipe(recipe)
+            .then((result) => {
+                const id = result.data?.library?.createRecipe.id;
+                id
```
Comment (barneyb) [inline src/features/RecipeEdit/RecipeAddController.tsx:45]: How could this `id` ever be false-y, other than if the new recipe was assigned ID `0`?

File: src/features/RecipeEdit/RecipeAddController.tsx (hunk 1 above)
Comment (switzerb) [reply src/features/RecipeEdit/RecipeAddController.tsx:15]: It sure would. That was silly of me. I shall change.

File: src/features/RecipeEdit/RecipeAddController.tsx (hunk 2 above)
Comment (switzerb) [reply src/features/RecipeEdit/RecipeAddController.tsx]: I was thinking that sometimes graphQL sends back a 200 and a partial response so it's better to be defensive...but honestly I have no idea if that is even possible in this instance so I can remove if unneeded.

Comment (barneyb) [reply src/features/RecipeEdit/RecipeAddController.tsx]: Wouldn't apollo have converted the "HTTP 200 w/ some errors in the JSON" into a promise rejection?

Comment (switzerb) [reply src/features/RecipeEdit/RecipeAddController.tsx]: That is a great question and I am realizing I am not up to speed on the apollo server error handling range. I removed this in favor of simplicity and now have on my to do list to read up on error handling for apollo server and client.

Comment (switzerb) [reply src/features/RecipeEdit/RecipeAddController.tsx]: I think this is the kind of thing that made me gun-shy of making assumptions: 

image

Comment (barneyb) [reply src/features/RecipeEdit/RecipeAddController.tsx]: Thank God we don't use Apollo Server!

**Review by barneyb — APPROVED** [review]

**Comment by barneyb** [comment]
This is going to need to merge into #97 not just `beta`, right?

**Comment by switzerb** [comment]
> This is going to need to merge into #97 not just `beta`, right?

yeah -- I've got a weird workflow happening here but I think it will work out. I have the two branches still on my local machine and so figured I'd cherrypick over the commits from PR #2 or something when I merge into main. It's maybe worth talking about a better workflow? The other option is to open a PR from beta -> main but I'm not sure that's a great idea.

**Pull request merged**